vllm/model_executor/layers/quantization/compressed_tensors/schemes/compressed_tensors_w8a8.py

```diff
@@ -75,3 +77,29 @@ def create_weights(self, layer: torch.nn.Module,
             "output_dim": 0,
             "weight_loader": weight_loader,
         })
+        
+        # INPUT SCALE
+        if self.input_dynamic:
+            # Dynamic quantization: set to None.
+            layer.input_scale = None
+        else:
+            # Static quantization:  load from disk.
+            input_scale = Parameter(torch.empty(1, dtype=torch.float32),
+                                requires_grad=False)
+            layer.register_parameter("input_scale", input_scale)
+            set_weight_attrs(input_scale, {
+                "weight_loader": weight_loader,
+                "ignore_warning": True,
+            })
+    
+    def apply_weights(self, layer: torch.nn.Module, x: torch.Tensor):
+        # ops.scaled_fp8_quant supports both dynamic and static quant.
+        # * dynamic, layer.input_scale is None and x_scale computed from x.
+        # * static, layer.input_scale is scalar and x_scale is input_scale.
+        x_q, x_scale = ops.scaled_int8_quant(x, layer.input_scale)
+
+        return ops.cutlass_scaled_mm(x_q,
+                                     layer.weight.t(), 
```

Comment:
Can you transpose the weight ahead of time?

Reply:
done!